site/docs/spark.md

```diff
@@ -728,3 +730,82 @@ spark.read.format("iceberg").load("db.table.files").show(truncate = false)
 // Hadoop path table
 spark.read.format("iceberg").load("hdfs://nn:8020/path/to/table#files").show(truncate = false)
 ```
+
+## Type compatibility
+
+Spark and Iceberg support different set of types. Iceberg does the type conversion automatically, but not for all combinations,
+so you may want to understand the type conversion in Iceberg in prior to design the types of columns in your tables.
+
+### Spark type to Iceberg type on creating table
+
+This type conversion table describes how Spark types are converted to the Iceberg types. The conversion applies on creating Iceberg table via Spark without using Iceberg core API.
+
+| Spark           | Iceberg                 | Notes |
+|-----------------|-------------------------|-------|
+| boolean         | boolean                 |       |
+| integer         | integer                 |       |
+| short           | integer                 |       |
+| byte            | integer                 |       |
+| long            | long                    |       |
+| float           | float                   |       |
+| double          | double                  |       |
+| date            | date                    |       |
+| timestamp       | timestamp with timezone |       |
+| string          | string                  |       |
+| char            | string                  |       |
+| varchar         | string                  |       |
+| binary          | binary                  |       |
+| decimal         | decimal                 |       |
+| struct          | struct                  |       |
+| array           | list                    |       |
+| map             | map                     |       |
+
+The type conversion is asymmetric: this table doesn't represent the types of Iceberg Spark can "read" from, or "write" to.
+The following sections describe the feasibility on read/write for Iceberg type from Spark.
+
+### Iceberg to Spark on reading from Iceberg table
+
+| Iceberg                    | Spark                   | Note  |
+|----------------------------|-------------------------|-------|
+| boolean                    | boolean                 |       |
+| integer                    | integer                 |       |
+| long                       | long                    |       |
+| float                      | float                   |       |
+| double                     | double                  |       |
+| date                       | date                    |       |
+| time                       | N/A                     |       |
+| timestamp with timezone    | timestamp               |       |
+| timestamp without timezone | N/A                     |       |
```

Comment:
Should we convert timestamp without zone to timestamp in Spark when reading?

Reply:
Honestly I wasn't too concerned about timestamp (I tried my best to consider the type as epoch based on UTC) so no strong opinion here. If we'd like to also support this (probably adjusting to the TZ in Spark?) it might be also valid to do vice versa, so that read and write are symmetric.